universe/Building.h

```diff
@@ -214,9 +214,8 @@ class BuildingTypeManager {
 
 private:
     BuildingTypeManager();
-    ~BuildingTypeManager();
 
-    std::map<std::string, BuildingType*> m_building_types;
+    std::map<std::string, std::unique_ptr<BuildingType>> m_building_types;
```

Comment:
This doesn't make sense to me. Shouldn't it be stored in the manager and returned from GetBuildingType as a shared_ptr? Storing as unique and then returning a raw pointer seems like a bad idea.

Returning from a parser as unique_ptr also seems pointless... why not just return a shared_ptr in the first place? It accomplishes the same thing (no leaks) without the awkwardness of having to convert from unique to shared for storage.